tooling/greybox_fuzzer/src/mutation/string.rs

```diff
@@ -0,0 +1,290 @@
+            self.prng.gen_range((first_chunk_position + 1)..=second_chunk_position);
+
+        let second_chunk_end = self.prng.gen_range((second_chunk_position + 1)..=buffer_length);
+
+        // Leave the start in place
+        result.extend_from_slice(&buffer[0..first_chunk_position]);
+
+        // Insert second chunk
+        result.extend_from_slice(&buffer[second_chunk_position..(second_chunk_end)]);
+
+        // Insert what's in between the chunks
+        result.extend_from_slice(&buffer[first_chunk_end..(second_chunk_position)]);
+
+        // Insert first chunk
+        result.extend_from_slice(&buffer[first_chunk_position..first_chunk_end]);
+
+        // Insert the tail
+        result.extend_from_slice(&buffer[second_chunk_end..buffer_length]);
+
+        result
+    }
+
+    /// Take a random chunk of the input and insert it several times into the input
+    fn duplicate_chunk(&mut self, input_buffer: &Vec<u8>) -> Vec<u8> {
+        let mut result = input_buffer.clone();
+        let buffer_length = input_buffer.len();
+        // The maximum length of the chunk is half the total length
+        let maximum_chunk_length = buffer_length / 2;
+
+        // Get a random position for the chunk
+        let chunk_position = self.prng.gen_range(0..=buffer_length - 1);
+
+        // Pick size
+        let chunk_size =
+            self.prng.gen_range(1..=min(buffer_length - chunk_position, maximum_chunk_length));
+
+        // Find an insertion position with enough space
+        let insertion_position = self.prng.gen_range(0..(buffer_length - chunk_size));
+
+        // Determine how many times to repeat
+        let maximum_insertion_count = (buffer_length - insertion_position) / chunk_size;
+        let insertion_count = self.prng.gen_range(0..=maximum_insertion_count);
+        for i in 0..insertion_count {
+            result.splice(
+                (insertion_position + i * chunk_size)..(insertion_position + (i + 1) * chunk_size),
+                input_buffer[chunk_position..(chunk_position + chunk_size)].iter().copied(),
+            );
+        }
+        result
+    }
+
+    /// Take a random value and insert it several times
+    fn duplicate_random_value(&mut self, input_buffer: &Vec<u8>) -> Vec<u8> {
+        let mut result = input_buffer.clone();
+        let buffer_length = input_buffer.len();
+
+        // Find an insertion position with enough space
+        let insertion_position = self.prng.gen_range(0..buffer_length);
+
+        // Pick count
+        let insertion_count = self.prng.gen_range(1..=(buffer_length - insertion_position));
+
+        // Pick value
+        let value = self.prng.gen_range(MIN_ASCII..=MAX_ASCII);
+        for item in result.iter_mut().skip(insertion_position).take(insertion_count) {
+            *item = value;
+        }
+        result
+    }
+    /// Create a spliced version of 2 buffers, where each element in the result is at the same index as in the original ones
```

Comment:
Could you include an example of this? That is, an example first_buffer and second_buffer, with one possible outcome.

I also think it would be good to include small examples like this in all mutations so it's easier to understand what they do without having to read the code.

Reply:
Added